InternalTimeSheet/app/src/main/java/br/com/accera/internaltimesheet/ui/main/MainPresenterImpl.java

```diff
@@ -1,26 +1,62 @@
 package br.com.accera.internaltimesheet.ui.main;
 
+import android.content.Intent;
 import android.text.TextUtils;
 import android.util.Log;
 
 import javax.inject.Inject;
 
+import br.com.accera.callback.DataCompleteResponse;
 import br.com.accera.core.presentation.ui.baseview.BasePresenterImpl;
+import br.com.accera.data.user.UserDto;
+import br.com.accera.data.user.UserRepository;
 import br.com.accera.internaltimesheet.R;
+import br.com.accera.internaltimesheet.ui.base.BaseTimesheetPresenter;
 
 /**
  * Created by LuisAlmeida on 25/05/18.
  */
 
-public class MainPresenterImpl extends BasePresenterImpl<MainContract.View> implements MainContract.Presenter{
+public class MainPresenterImpl extends BaseTimesheetPresenter<MainContract.View> implements MainContract.Presenter{
+
+    private UserRepository userRepository;
 
     @Inject
-    public MainPresenterImpl() {
+    public MainPresenterImpl(UserRepository userRepository) {
+        this.userRepository = userRepository;
     }
 
     public void receiveClick(String nome, String diainit, String jornadainit,  String diaend, String jornadaend){
         if (TextUtils.isEmpty(nome)){
             mView.getAlertHelper().showErrorMessage(R.string.no_name);
         }
+        if (TextUtils.isEmpty(diainit)){
```

Comment:
You can create a StringBuilder to concat all errors. It might be more useful than show many error alerts.